### PR TITLE
ref(helm): make FakeReleaseClient public

### PR DESCRIPTION
**Pull request by michelleN**
so it can be used by other projects that use the helm client

*also renamed to FakeClient

### CODE REVIEW

**Comment by bacongobbler** [comment]
LGTM from an external standpoint. Any reason we can't use the real helm client with the storage layer set to the filesystem rather than configmaps, or does the fakeclient also "fake out" the connection to kubernetes and just store release information in-memory?

**Comment by adamreese** [comment]
could this be made a subpackage?  `pkg/helm/fake`

File: pkg/helm/fake.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+type FakeClient struct {
+	Rels      []*release.Release
+	Responses map[string]release.TestRun_Status
+	Err       error
+}
+
+var _ Interface = &FakeClient{}
+var _ Interface = (*FakeClient)(nil)
+
+// Listreleases lists the current releases
```
Comment (thomastaylor312) [inline pkg/helm/fake.go]: Nit: Spelling. `ListReleases`

**Review by thomastaylor312 — APPROVED** [review]
This all looks good from a code perspective (with one super tiny nit). I agree with @adamreese though that I think this would be better in its own subpackage `pkg/helm/fake`, thoughts?. However, I am not going to hold up merging on that point. If you like this how it is, feel free to merge.

File: pkg/helm/fake.go (hunk 1 above)
Comment (michelleN) [inline pkg/helm/fake.go]: good catch. I'll fix that.

**Comment by michelleN** [comment]
@adamreese @thomastaylor312 I can't put fake in a sub package mainly because I need things from `pkg/helm/option.go`
Another option is to pull it out and put it in something like `pkg/testing`

I could potentially fake out everything in that option file but it seems like overkill to me. Thoughts?

**Comment by thomastaylor312** [comment]
@michelleN So I know there is a lot of testing stuff we want to improve (like integration tests!), what if we merge this as is and create an issue to track that we want to make things even cleaner with a testing package and a meta-issue to track all the testing improvements we want. Does that sound like it would work?

**Comment by michelleN** [comment]
@thomastaylor312 That makes sense to me but at that point we'd be changing public functions. So the refactor into a testing package would have to wait till helm 3.0.0 if I'm not mistaken. If we're cool with that, I'm good with your plan 👍

**Comment by thomastaylor312** [comment]
That is ok by me. Feel free to merge!

**Review by adamreese — APPROVED** [review]

**Pull request merged**